judge.pl

```diff
@@ -221,6 +223,9 @@ sub generate_test_group {
         $_->{generated} = 1;
         my $tf = $problem_cache->test_file($pid, $_);
         $fu->copy_glob([ $cfg->rundir, sprintf($out, $_->{rank}) ], $tf) or return;
+        if (!defined $test->{in_file}) {
```

Comment:
You are NOT checking hashes for non-generated files. Why?

Reply:
But the issue is about generated input. Anyway, what's the reason to check it for non-generated files?

Reply:
Consistency at least. It would be unexpected to user if
`<Test><In hash="$sha1$zzz">123</In></Test>`
does not raise error.
"Never add extra code just to make user's life worse" is my usual principle.